### PR TITLE
feat: implement habit deletion

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR
Added the ability to delete habits from the app through both the habit card menu and edit screen.

https://github.com/user-attachments/assets/2c96de4c-2aea-4dfd-9434-934fc8bc9d26

### What changed?
- Created a new `useDeleteHabit` mutation hook for handling habit deletion
- Added a delete button to the habit edit screen with confirmation dialog
- Implemented delete functionality in the habit card's menu
- Fixed habit order management to properly handle habit deletion
- Updated mock habits implementation to correctly handle state updates

### How to test?
1. Create a new habit
2. Try deleting it through:
   - The three-dot menu on the habit card
   - The delete button on the edit screen
3. Verify that:
   - A confirmation dialog appears when deleting from the edit screen
   - The habit is removed from the list
   - The habit order is maintained correctly
   - A success toast message appears after deletion

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/148?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#148** <a href="https://app.graphite.dev/github/pr/habits-together/app/148?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/148?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#147** <a href="https://app.graphite.dev/github/pr/habits-together/app/147?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#146** <a href="https://app.graphite.dev/github/pr/habits-together/app/146?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#145** <a href="https://app.graphite.dev/github/pr/habits-together/app/145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#144** <a href="https://app.graphite.dev/github/pr/habits-together/app/144?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]

**Review by AnkushSarkar10 — APPROVED** [review]

**Comment by owengretzinger** [comment]
### Merge activity

* **Jan 10, 4:47 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/habits-together/app/148).
